packages/@react-stately/selection/src/useMultipleSelectionState.ts

```diff
@@ -65,13 +75,45 @@ export function useMultipleSelectionState(props: MultipleSelectionStateProps): M
     props.disabledKeys ? new Set(props.disabledKeys) : new Set<Key>()
   , [props.disabledKeys]);
   let [selectionBehavior, setSelectionBehavior] = useState(props.selectionBehavior || 'toggle');
+  let updateFocusedKey = (k, childFocusStrategy = 'first') => {
+    focusedKeyRef.current = k;
+    childFocusStrategyRef.current = childFocusStrategy;
+    setFocusedKey(k);
+  };
 
   // If the selectionBehavior prop is set to replace, but the current state is toggle (e.g. due to long press
   // to enter selection mode on touch), and the selection becomes empty, reset the selection behavior.
   if (props.selectionBehavior === 'replace' && selectionBehavior === 'toggle' && typeof selectedKeys === 'object' && selectedKeys.size === 0) {
     setSelectionBehavior('replace');
   }
 
+  let lastSelectedKeysProp = useRef(selectedKeysProp);
+  let focusedKeySelected = selectedKeys === 'all' || selectedKeys.has(focusedKeyRef.current);
+  // If the collection doesn't have focus and the selected keys change, move focus to the first selected key.
+  // Only do this selection is controlled and there is something actually selected (not empty set)
+  if (
+    collection &&
+    selectedKeysProp != null &&
+    (selectedKeysProp === 'all' || selectedKeysProp.size > 0) &&
+    !isFocusedRef.current &&
+    !focusedKeySelected &&
+    !equalSets(selectedKeysProp, lastSelectedKeysProp.current)
+  ) {
+    let firstSelectedKey = null;
+    if (selectedKeysProp === 'all') {
+      firstSelectedKey = collection.getFirstKey();
+    } else {
+      for (let key of selectedKeysProp) {
+        let item = collection.getItem(key);
+        if (!firstSelectedKey || item?.index < firstSelectedKey.index) {
+          firstSelectedKey = item.key;
+        }
+      }
```

Comment:
Logic taken from SelectionManager. Can't pass a SelectionManager into this hook cuz the SelectionManager needs the state from this hook

Reply:
I'm unsure about this, it makes sense for flat structures, Table/TabList/etc, but I'm not sure it makes sense with trees. We could cause very long series of branches to open by changing the selected keys.

Should this be handled per component? or maybe we have a hook that does it for flat and one that does it for trees?

Reply:
Maybe this is related to this conversation. I wasn't following why we are using multiple selection, I thought we could only ever have one tab selected.

Reply:
The initial thought was that this wasn't a tabs specific issue, but rather this could happen in any of our components that support selection and rely on the state tracked `focusedKey` to determine what tab index goes where. Gonna explore moving it up a level to the component specific state hooks